### PR TITLE
Force using tuples in frozenset constructor

### LINKED ISSUE #920: Force using tuples in frozenset constructor
# Rule request

## Thesis

Force using tuples in frozenset constructor. For example,

```python
a = frozenset([2])
```
should be rewritten as

```python
a = frozenset((2,))
```

## Reasoning
Immutable types in constructors are more consistent then mutable

### PR DESCRIPTION
**Pull request by slapshin**
## Checklist

- [x] I have double-checked that there are no unrelated changes in this pull request (old patches, accidental config files, etc)
- [x] I have created at least one test case for the changes I have made
- [x] I have updated the documentation for the changes I have made
- [x] I have added my changes to the `CHANGELOG.md`

## Related issues
- Closes #920

I have added `call_args.get_all_args` for eliminating code duplication and adjust the code. Also, some refactoring in `function_args.py`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -30,6 +30,7 @@ We used to have incremental versioning before `0.1.0`.
 - Forbids to start lines with `.`
 - Enforces better `&`, `|`, `>>`, `<<`, `^` operators usage
 - Forbids incorrect exception order
+- `WPS527` force using tuples as method arguments
```
Comment (sobolevn) [inline CHANGELOG.md]: Let's say: `Enforces tuples usage with frozenset constructor`

File: tests/test_visitors/test_ast/test_statements/test_force_tuple_arguments.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    ForceTupleArgumentsViolation,
+)
+from wemake_python_styleguide.visitors.ast.statements import (
+    WrongMethodParametersVisitor,
+)
+
+
+@pytest.mark.parametrize('code', [
+    'a = {0}((1,))',
+    'a = {0}((1,), b)',
```
Comment (sobolevn) [inline tests/test_visitors/test_ast/test_statements/test_force_tuple_arguments.py]: Let's also test: 
- `{0}(())`
- `{0}((1, 2, 3))`

File: tests/test_visitors/test_ast/test_statements/test_force_tuple_arguments.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    tree = parse_ast_tree(mode(code.format(method)))
+
+    visitor = WrongMethodParametersVisitor(default_options, tree=tree)
+    visitor.run()
+
+    assert_errors(visitor, [])
+
+
+@pytest.mark.parametrize('code', [
+    'a = {0}([1])',
```
Comment (sobolevn) [inline tests/test_visitors/test_ast/test_statements/test_force_tuple_arguments.py]: What about comprehensions? Should we also test these cases?

File: wemake_python_styleguide/constants.py [hunk 4]
```diff
@@ -359,3 +359,9 @@
 # Used to specify a pattern which checks variables and modules for underscored
 # numbers in their names:
 UNDERSCORED_NUMBER_PATTERN: Final = re.compile(r'.+\D\_\d+(\D|$)')
+
+# List of functions in which parameters must be of type tuple instead
+# another iterables
+TUPLE_ARGUMENTS_METHODS = frozenset((
```
Comment (sobolevn) [inline wemake_python_styleguide/constants.py]: This is a public constant. It should be moved to the upper part of the file and documented using `#:`

File: wemake_python_styleguide/violations/refactoring.py [hunk 5]
```diff
@@ -42,6 +42,7 @@
    MisrefactoredAssignmentViolation
    InCompareWithSingleItemContainerViolation
    ImplicitYieldFromViolation
+   ForceTupleArgumentsViolation
```
Comment (sobolevn) [inline wemake_python_styleguide/violations/refactoring.py]: `NotATupleArgumentViolation` is a better name.

File: wemake_python_styleguide/violations/refactoring.py [hunk 6]
```diff
@@ -1046,3 +1048,35 @@ class ImplicitYieldFromViolation(ASTViolation):
...
+
+    See
+    :py:data:`~wemake_python_styleguide.constants.TUPLE_ARGUMENTS_METHODS`
+    for full list of methods that we check for.
+
+    .. versionadded:: 0.13.0
+
+    """
+
+    error_template = 'Found missing using tuples as method arguments'
```
Comment (sobolevn) [inline wemake_python_styleguide/violations/refactoring.py]: `'Found not a tuple used as an argument'`

File: wemake_python_styleguide/visitors/ast/statements.py [hunk 7]
```diff
@@ -457,3 +460,38 @@ def _check_augmented_assign_pattern(
...
+    """Ensures that all parameters follow our rules."""
+
+    _no_tuples_collections: ClassVar[AnyNodes] = (
+        ast.List,
+        ast.Set,
+    )
+
+    def visit_Call(self, node: ast.Call) -> None:
+        """Checks call arguments indentation."""
+        all_args = call_args.get_all_args(node)
```
Comment (sobolevn) [inline wemake_python_styleguide/visitors/ast/statements.py]: Let's move this line into `_check_tuple_parameters_types`

File: tests/test_visitors/test_ast/test_statements/test_force_tuple_arguments.py (hunk 3 above)
Comment (slapshin) [reply tests/test_visitors/test_ast/test_statements/test_force_tuple_arguments.py]: hm... agree!

File: wemake_python_styleguide/visitors/ast/statements.py (hunk 7 above)
Comment (slapshin) [reply wemake_python_styleguide/visitors/ast/statements.py]: sorry, don't understand

**Comment by slapshin** [comment]
Thanks for editing

**Comment by sobolevn** [comment]
Done! Thanks a lot for writing this. I really appreciate your time and effort.

As a result we have a nice new rule right here! 👍

**Pull request merged**